### PR TITLE
Add support discover tags for multi-branch pipeline

### LINKED ISSUE #3051: feature: support tag trigger pipeline
/area devops
/milestone 3.1.0
link to https://kubesphere.com.cn/forum/d/2578-gitlabtag

### PR DESCRIPTION
**Pull request by LinuxSuRen**
**What type of PR is this?**
/kind feature

**What this PR does / why we need it**:
Add support tag discover for multi-branch pipeline

**Which issue(s) this PR fixes**:
Fixes #3051

**Special notes for reviewers**:

**Additional documentation, usage docs, etc.**:

### CODE REVIEW

File: pkg/controller/pipeline/pipeline_controller.go [hunk 1]
```diff
@@ -239,6 +239,8 @@ func (c *Controller) syncHandler(key string) error {
 					klog.V(8).Info(err, fmt.Sprintf("failed to update pipeline config %s ", key))
 					return err
 				}
+			} else {
+				klog.V(7).Info(fmt.Sprintf("nothing was changed, pipeline '%v'", copyPipeline.Spec))
```
Comment (shaowenchen) [inline pkg/controller/pipeline/pipeline_controller.go:243]: The level of the log needs to be changed.

Comment (LinuxSuRen) [reply pkg/controller/pipeline/pipeline_controller.go:243]: Got you.

**Comment by weekyuan** [comment]
How can I experience this new feature on 3.0?

**Comment by LinuxSuRen** [comment]
hi @weekyun, of course. Welcome! But please make sure DO NOT test it in any official environment yet. I highly suggest that you test it in a personal or temporary environment. Because it has not been released yet.

OK. I guess you're all ready to go. Let's rock it. There're four components you need to update:

* ks-apiserver  --> `kubespheredev/ks-apiserver`
* ks-controller-manager  -->  `kubespheredev/ks-controller-manager`
* ks-console  -->  `kubespheredev/ks-console`
* ks-jenkins  -->  `kubespheredev/ks-jenkins:2.249.1`

After you updated these deployments with the latest container images. You can create a multi-branch pipeline with Gitlab. But there's still a couple of things you might need to know.

# Restriction

* You need to add a Gitlab server on the global configuration page of Jenkins manually
* You need to type the Gitlab server on Kubersphere, please make sure you type the exact same string
* You can only use the [public service of Gitlab](https://gitlab.com/) instead of a private Gitlab until kubesphere/console#1361

**Comment by LinuxSuRen** [comment]
hi @weekyun, of course. Welcome! But please make sure DO NOT test it in any official environment yet. I highly suggest that you test it in a personal or temporary environment. Because it has not been released yet.

OK. I guess you're all ready to go. Let's rock it. There're four components you need to update:

* ks-apiserver  --> `kubespheredev/ks-apiserver`
* ks-controller-manager  -->  `kubespheredev/ks-controller-manager`
* ks-console  -->  `kubespheredev/ks-console`
* ks-jenkins  -->  `kubespheredev/ks-jenkins:2.249.1`

After you updated these deployments with the latest container images. You can create a multi-branch pipeline with Gitlab. But there's still a couple of things you might need to know.

# Restriction

* You need to add a Gitlab server on the global configuration page of Jenkins manually
* You need to type the Gitlab server on Kubersphere, please make sure you type the exact same string
* You can only use the [public service of Gitlab](https://gitlab.com/) instead of a private Gitlab until kubesphere/console#1361

**Comment by LinuxSuRen** [comment]
Hi @weekyuan. Sorry for misleading you about this. It looks like the front-end is not ready. Please follow https://github.com/kubesphere/console/issues/1310

**Comment by weekyuan** [comment]
> Hi @weekyuan. Sorry for misleading you about this. It looks like the front-end is not ready. Please follow [kubesphere/console#1310](https://github.com/kubesphere/console/issues/1310)

okay,Thank you for your help!

**Pull request merged**